File: .skills/openclaw-skills/skills/wdmcygah/chinese-image-gen/scripts/render.py
#!/usr/bin/env python3
"""
中文竖屏图片生成器
用法: python3 render.py <html_file> <output.png> [width] [height]
"""
import sys
import time
from playwright.sync_api import sync_playwright

def render(html_content, output_path, width=750, height=1334):
    with sync_playwright() as p:
        browser = p.chromium.launch(args=['--no-sandbox'])
        page = browser.new_page(viewport={"width": width, "height": height})
        page.set_content(html_content)
        page.wait_for_load_state("networkidle")
        time.sleep(1.5)  # 等 Google Fonts 加载
        page.screenshot(path=output_path)
        page.close()
        browser.close()
        print(f"✅ {output_path}")

if __name__ == "__main__":
    if len(sys.argv) < 3:
        print("用法: python3 render.py <html_file> <output.png> [width] [height]")
        sys.exit(1)
    
    html_file = sys.argv[1]
    output = sys.argv[2]
    w = int(sys.argv[3]) if len(sys.argv) > 3 else 750
    h = int(sys.argv[4]) if len(sys.argv) > 4 else 1334
    
    with open(html_file, 'r') as f:
        html = f.read()
    
    render(html, output, w, h)

File: .skills/openclaw-skills/skills/tangyijun/nuomi-xhs-card/scripts/src/core/render.ts
import { mkdirSync } from 'node:fs';
import { dirname, resolve } from 'node:path';
import { chromium } from 'playwright';
import { buildCardDocument } from './template.js';
import type { RenderResult, ThemeSpec } from './types.js';

interface ExportPagesArgs {
  pages: string[];
  theme: ThemeSpec;
  width: number;
  height: number;
  scale: number;
  showPager: boolean;
  outputDir: string;
  title?: string;
  mainTitle?: string;
}

export async function exportPagesToPng(args: ExportPagesArgs): Promise<RenderResult[]> {
  mkdirSync(args.outputDir, { recursive: true });

  const browser = await chromium.launch();
  const page = await browser.newPage({
    viewport: { width: args.width, height: args.height },
    deviceScaleFactor: args.scale
  });

  const total = args.pages.length;
  const results: RenderResult[] = [];

  try {
    for (let index = 0; index < total; index += 1) {
      const pageNumber = index + 1;
      const documentHtml = buildCardDocument({
        theme: args.theme,
        contentHtml: args.pages[index],
        width: args.width,
        height: args.height,
        pageNum: `${pageNumber} / ${total}`,
        showPager: args.showPager,
        title: index === 0 ? args.title : undefined,
        mainTitle: index === 0 ? args.mainTitle : undefined
      });

      const outputPath = resolve(args.outputDir, `card_${pageNumber}.png`);
      await page.setContent(documentHtml, { waitUntil: 'load' });
      await page.waitForTimeout(30);
      await page.screenshot({
        path: outputPath,
        type: 'png',
        clip: { x: 0, y: 0, width: args.width, height: args.height },
        animations: 'disabled'
      });

      results.push({
        index: pageNumber,
        imagePath: outputPath,
        success: true
      });
    }
  } finally {
    await browser.close();
  }

  return results;
}

export async function doctorPlaywright(): Promise<{ ok: boolean; message: string }> {
  try {
    const browser = await chromium.launch();
    const page = await browser.newPage({ viewport: { width: 200, height: 100 } });
    await page.setContent('<html><body><div>ok</div></body></html>', { waitUntil: 'load' });
    await browser.close();
    return { ok: true, message: 'Playwright Chromium is ready.' };
  } catch (error) {
    return {
      ok: false,
      message: `Playwright Chromium unavailable: ${error instanceof Error ? error.message : String(error)}`
    };
  }
}

export function ensureOutputDir(pathValue: string): string {
  const outputDir = resolve(pathValue);
  mkdirSync(outputDir, { recursive: true });
  return outputDir;
}

export function ensureParentDir(pathValue: string): void {
  mkdirSync(dirname(pathValue), { recursive: true });
}
